### PR TITLE
Cce analytic test spherical metric

### PR DESCRIPTION
**Pull request by moxcodes**
## Proposed changes

Adds another layer of abstract base class, this time for data that is most easily represented in a spherical coordinate system.

### Types of changes:

- [x] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### Dependencies
- [x] #2233

### CODE REVIEW

**Comment by moxcodes** [comment]
Force pushed a quick clang-tidy fix.

**Comment by moxcodes** [comment]
Rebased on develop, no longer dependent.
Cheers!

File: src/Evolution/Systems/Cce/AnalyticSolutions/SphericalMetricData.hpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+  /*!
+   * \brief Computes the Cartesian spacetime metric from the spherical solution
+   * provided by the derived classes.
+   *
+   * \details The derived classes provide spherical metric data via the virtual
+   * function `SphericalMetricData::spherical_metric()` at a resolution
+   * determined by member variable `l_max_`. This function performs the
+   * coordinate transformation using the Jacobian computed from
+   * `SphericalMetricData::inverse_jacobian()` and performs a re-grid to the
```
Comment (kidder) [inline src/Evolution/Systems/Cce/AnalyticSolutions/SphericalMetricData.hpp]: what is the reason for re-gridding? (I.e. why are there two resolutions?)

File: src/Evolution/Systems/Cce/AnalyticSolutions/SphericalMetricData.cpp [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+        intermediate_d_cartesian_metric.get(i, a, b) =
+            intermediate_jacobian.get(i, 0) *
+                intermediate_dr_cartesian_metric.get(a, b) -
+            intermediate_jacobian.get(i, 1) * real(derivative_buffer.data()) -
+            intermediate_jacobian.get(i, 2) * imag(derivative_buffer.data());
+      }
+    }
+  }
+
+  // resample each component.
```
Comment (kidder) [inline src/Evolution/Systems/Cce/AnalyticSolutions/SphericalMetricData.cpp]: and if you keep the resampling, can't it call a generic function to resample each component?

**Review by kidder — CHANGES_REQUESTED** [review]
reviewed with @nilsdeppe

File: src/Evolution/Systems/Cce/AnalyticSolutions/SphericalMetricData.hpp (hunk 1 above)
Comment (moxcodes) [reply src/Evolution/Systems/Cce/AnalyticSolutions/SphericalMetricData.hpp]: Sometimes the internal computations, for instance the robinson-trautman solution, or the angular derivatives needed to assemble the spatial derivatives of the spatial metric, are operations that are sensitive to the angular resolution.
So, my reasoning was that calling code might prefer to have a different resolution for the boundary calculation than for the main evolution, though now that you point it out, it probably makes the interface more confusing and gives only marginal gains in efficiency.
tl;dr: I'll remove the resampling. :P

**Comment by moxcodes** [comment]
pushed a fixup to remove the regrid operation.
Cheers!

**Review by nilsdeppe — COMMENTED** [review]
I'm happy for you to squash once @kidder has looked at it

**Comment by kidder** [comment]
yes, squash

**Comment by moxcodes** [comment]
fixup squashed.
Cheers!

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]
